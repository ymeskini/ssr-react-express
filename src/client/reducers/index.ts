import { combineReducers } from 'redux';

const authentication = (
  state = {
    isLogged: false
  }
) => {
  return state;
};

export type State = {
  authentication: typeof authentication;
};

export const rootReducer = combineReducers({ authentication });
