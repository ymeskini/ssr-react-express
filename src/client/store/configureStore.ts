import { createStore, Store } from 'redux';
import { createBrowserHistory, createMemoryHistory } from 'history';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import { rootReducer } from '../reducers';

export const history = process.env.IS_BROWSER ? createBrowserHistory() : createMemoryHistory();

export const configureStore = (
  preloadedState: Record<string, any> = {}
): {
  store: Store;
} => {
  const store = createStore(rootReducer, preloadedState, composeWithDevTools());

  /* istanbul ignore next */
  if (module.hot) {
    module.hot.accept('../reducers', async () => {
      const { rootReducer } = await import(/* webpackMode: "eager" */ '../reducers');

      store.replaceReducer(rootReducer);
    });
  }

  return { store };
};
