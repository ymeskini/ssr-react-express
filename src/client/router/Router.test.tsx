import * as React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import { rootReducer } from '../reducers';
import { Router } from '.';

let state: any;
let store: any;

jest.mock('./routes.ts', () => {
  return {
    LoadableHomePage: () => <p>top</p>
  };
});

beforeEach(() => {
  state = createStore(rootReducer).getState();
  store = configureStore()(state);
});

test('should render Top page', () => {
  const tree = render(
    <Provider store={store}>
      <MemoryRouter initialEntries={[{ pathname: '/', key: 'testKey' }]}>
        <Router />
      </MemoryRouter>
    </Provider>
  );

  expect(tree.asFragment()).toMatchSnapshot();
});
