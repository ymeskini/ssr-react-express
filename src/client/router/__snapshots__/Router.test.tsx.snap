// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Top page 1`] = `
<DocumentFragment>
  <p>
    top
  </p>
</DocumentFragment>
`;
